src/ddmd/intrange.d

```diff
@@ -77,35 +78,70 @@ struct SignExtendedNumber
             return 0;
     }
 
-    SignExtendedNumber opNeg() const
+    private alias SignExtendedNumber R;
+
+    R opUnary(string op : "++")()
+    {
+        if (value != UINT64_MAX)
+            ++value;
+        else if (negative)
+        {
+            value = 0;
+            negative = false;
+        }
+        return this;
+    }
+
+    R opUnary(string op : "~")() const
+    {
+        return SignExtendedNumber(~value);
```

Comment:
I'm not sure this (and other similar operations) is correct. It converts a negative number into a positive one. Shouldn't it be `return SignExtendedNumber(~value, negative);`? cc @WalterBright @tgehr

Reply:
``~`` flips the sign bit, so if might need to be ``SignExtendedNumber(~value, !negative)`` if the underlying datatype is signed.

Reply:
@tgehr I actually tried that, then druntime failed to compile